### PR TITLE
Wait for ephemeral container

### LINKED ISSUE #60: Wait for ephemeral container to be ready
Trying to execute a command in an ephemeral container immediately after creating it may fail with the error `ERRO[0000] GoError: unable to upgrade connection: container not found ("k6-chaos")` . It would be convenient adding an option for waiting the ephemeral container to be ready.

### PR DESCRIPTION
**Pull request by pablochacin**
Add option for waiting for ephemeral container to be running

closes #60

### CODE REVIEW

File: pkg/pods/pods.go [hunk 1]
```diff
@@ -278,7 +303,7 @@ func (obj *Pods) Exec(options ExecOptions) (*ExecResult, error) {
 
 // AddEphemeralContainer adds an ephemeral container to a running pod. The Pod is identified by name and namespace.
 // The container is described by options
-func (obj *Pods) AddEphemeralContainer(name, namespace string, options ContainerOptions) error {
+func (obj *Pods) AddEphemeralContainer(name, namespace string, options EphemeralContainerOptions) error {
```
Comment (javaducky) [inline pkg/pods/pods.go:306]: API-breaking change for clients. Since it's early in the process for ephemeral containers, we should be fine.

**Review by javaducky — CHANGES_REQUESTED** [review]
Recommend using [Act](https://github.com/nektos/act) to run linter locally. There are some additional linting failures to cause your changes to fail upon merge.

`act --job run-golangci`

We should probably update our Github Actions to run the `run-golangci` job on creation of PRs rather than upon merge to `main`.

File: pkg/pods/pods.go [hunk 2]
```diff
@@ -80,6 +85,8 @@ type PodOptions struct {
 	Wait          string                 // timeout for waiting until the pod is running
 }
 
+type PodConditionChecker func(*k8sTypes.Pod) (bool, error)
```
[the remark this answers is not in the document]
Comment (pablochacin) [reply pkg/pods/pods.go:88]: No it doesn't would change it to private to the package

File: pkg/pods/pods.go [hunk 3]
```diff
@@ -61,6 +61,11 @@ type ContainerOptions struct {
 	Capabilities []string // capabilities to be added to the container's security context
 }
 
+type EphemeralContainerOptions struct {
```
[the remark this answers is not in the document]
Comment (pablochacin) [reply pkg/pods/pods.go:64]: Done

File: pkg/pods/pods.go [hunk 4]
```diff
@@ -194,10 +201,30 @@ func (obj *Pods) Wait(options WaitOptions) (bool, error) {
...
+			}
+			if string(pod.Status.Phase) == options.Status {
+				return true, nil
+			}
+			return false, nil
+		},
+	)
+}
+
+// waitForCondition watchs a Pod in a given namespace until a condition is validated by a PodConditionChecket or a timeout expires.
```
[the remark this answers is not in the document]
Comment (pablochacin) [reply pkg/pods/pods.go:221]: Fixed

**Review by javaducky — APPROVED** [review]
Verified changes will now pass linting. LGTM!

**Pull request merged**